spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/QueryMapper.java

```diff
@@ -402,10 +403,16 @@ private Condition createCondition(Column column, @Nullable Object mappedValue, i
 		}
 
 		if (comparator == Comparator.IS_TRUE) {
+			if (dialect instanceof SqlServerDialect) {
```

Comment:
Anything that is database specific needs to go into the `Dialect`. Instanceof checks for the `Dialect` implementation are a hard no-go.

I'm wondering if a better way to handle this is to register a converter for booleans that we then apply to the value.

Reply:
I will introduce a new default method on `Dialect` interface to handle boolean values.